ETL/main.py

```diff
@@ -0,0 +1,196 @@
+        ','.join(data.columns) + ") VALUES "
+    allrowvalues = []
+    for index, row in data.iterrows():
+        currentrowvalues = []
+        for col in data.columns:
+            # Get the values for the current row and sanitize them into clean string data
+            currentrowvalues.append(
+                "'" + str(row[col]).replace("'", "").replace("[", "").replace("]", "").replace("nan", "") + "'")
+        allrowvalues.append(','.join(currentrowvalues))
+    for item in allrowvalues:
+        sql += "({})".format(item)
+        if item != list(allrowvalues)[-1]:
+            sql += ','
+    cursor.execute(sql)
+    db.commit()
+
+
+log("Python ETL Script Start")
+
+# --------- PHASE 1: Import from Airtables --------- #
+
+log("Import the listings airtable")
+listings_airtable = Airtable(AIRTABLE_BASE_ID, 'listings', AIRTABLE_API_KEY)
+listings = listings_airtable.get_all()
+create_import_table('etl_import_1', listings)
+del listings_airtable
+del listings
+log("Import the phone airtable")
+phone_airtable = Airtable(AIRTABLE_BASE_ID, 'phone', AIRTABLE_API_KEY)
+phone = phone_airtable.get_all()
+create_import_table('etl_import_2', phone)
+del phone_airtable
+del phone
+log("Import the address airtable")
+address_airtable = Airtable(AIRTABLE_BASE_ID, 'address', AIRTABLE_API_KEY)
+address = address_airtable.get_all()
+create_import_table('etl_import_3', address)
+del address_airtable
+del address
+log("Import the contacts airtable")
+contacts_airtable = Airtable(
+    AIRTABLE_BASE_ID, 'contacts', AIRTABLE_API_KEY)
+contacts = contacts_airtable.get_all()
+create_import_table('etl_import_4', contacts)
+del contacts_airtable
+del contacts
+log("Import the contacts airtable")
+parent_airtable = Airtable(
+    AIRTABLE_BASE_ID, 'parent_organization', AIRTABLE_API_KEY)
+parent = parent_airtable.get_all()
+create_import_table('etl_import_5', parent)
+del parent_airtable
+del parent
+
+# --------- PHASE 2: Merge multiple tables into a single staging table --------- #
+
+log("Merge the import tables")
+query = ("select etl_merge_import_tables();")
+cursor.execute(query)
+db.commit()
+
+# --------- PHASE 3: Geocode addresses --------- #
+
+log("Geocode addresses")
+geolocator = Nominatim(user_agent="rose-city-resource")
+query = "SELECT * FROM etl_staging_1;"
+cursor.execute(query)
+columns = []
+for column in cursor.description:
+    columns.append(column.name)
+i_lat = columns.index('lat')
+i_lon = columns.index('lon')
+i_id = columns.index('id')
+i_full_address = columns.index('full_address')
+rows = cursor.fetchall()
+for row in rows:
+    address = row[i_full_address]
+    if address == None or address == '':
+        continue
+    location = geolocator.geocode(address)
+    lat = ''
+    lon = ''
+    try:
+        lat = location.latitude
+    except:
+        lat = ''
+    try:
+        lon = location.longitude
+    except:
+        lon = ''
+    sql = "UPDATE etl_staging_1 SET lat='{0}', lon='{1}' WHERE id='{2}';".format(
```

Comment:
Personally a fan of f-strings but that is just a suggestion.

Reply:
again just a suggestion.

Reply:
Thanks! I didn't know about f-strings, and now I'm gonna use 'em all the time! Updated the script.